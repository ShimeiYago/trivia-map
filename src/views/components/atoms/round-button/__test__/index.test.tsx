import { shallow, ShallowWrapper } from 'enzyme';
import { RoundButton, Props } from '..';

let wrapper: ShallowWrapper;

const props: Props = {
  children: 'button',
  onClick: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<RoundButton {...props} />);
  });

  it('base', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('selected', () => {
    wrapper.setProps({ selected: true });
    expect(wrapper).toMatchSnapshot();
  });
});
