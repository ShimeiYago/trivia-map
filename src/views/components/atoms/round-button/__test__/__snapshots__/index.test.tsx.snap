// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests base 1`] = `
<button
  className="round-button"
  onClick={[MockFunction]}
>
  <ForwardRef(Typography)
    color="black"
    component="div"
    variant="button"
  >
    button
  </ForwardRef(Typography)>
</button>
`;

exports[`Shallow Snapshot Tests selected 1`] = `
<button
  className="round-button selected"
  onClick={[MockFunction]}
>
  <ForwardRef(Typography)
    color="white"
    component="div"
    variant="button"
  >
    button
  </ForwardRef(Typography)>
</button>
`;
